### ISSUE TITLE
Docker error mariadb after update to 2..3.0.2

### ISSUE THREAD

**Issue by tiagoheineck** [docker/for-win#6658]
After update to 2.3.0.2 docker container cannot working correctly

# Expected behavior

- create folder db em run mariadb container with docker-compose up

### Actual behavior

- database container exited with code 1

### Information
My docker-compose file 

```
db:
    image: mariadb:10.3.22
    ports: 
    - "3306:3306"
    environment: 
        MYSQL_ROOT_PASSWORD: root
        MYSQL_DATABASE: root
        MYSQL_USER: root
        MYSQL_PASSWORD: root
    volumes:
    - ./db:/var/lib/mysql  
```

I tried other ways to find it here, but none works correctly, before the update was working like this perfectly, tested on another windows 10 machine, with docker and the same problem occurs

without specifying the volume works correctly

LOG ERROR: 
2020-05-12 14:41:31 0 [ERROR] InnoDB: preallocating 12582912 bytes for file ./ibdata1 failed with error 38
db_1   | 2020-05-12 14:41:31 0 [ERROR] InnoDB: Could not set the file size of './ibdata1'. Probably out of disk space
db_1   | 2020-05-12 14:41:31 0 [ERROR] InnoDB: Database creation was aborted with error Generic error. You may need to delete the ibdata1 file before trying to start up again.
db_1   | 2020-05-12 14:41:32 0 [ERROR] Plugin 'InnoDB' init function returned error.
db_1   | 2020-05-12 14:41:32 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.

  - Windows Version: 10
  - Docker Desktop Version: 2.3.0.2

### Steps to reproduce the behavior
1. Open Powershell and Create a directory "mkdir teste"
2. Create docker-compose.yml
3. Paste

```
version: '3.3'
services:
  db:
    image: mariadb:10.3.22
    ports: 
    - "3306:3306"
    environment: 
        MYSQL_ROOT_PASSWORD: covid
        MYSQL_DATABASE: plataformasaude
        MYSQL_USER: plataformasaude
        MYSQL_PASSWORD: covid     
    volumes:
    - ./db:/var/lib/mysql  
```

4. Run docker-compose up

**Comment by simonferquel** [comment]
Is it using the WSL2 backend or the hyper-v one ?

**Comment by stephen-turner** [comment]
The error message says that you're out of disk space. Is that in fact the case? You could try `docker image prune` or `docker system prune` to clean up your disk space.

**Comment by svacher** [comment]
Hi,

I'have the same problem after update too.
Not a real free disk space.

Someone can explain me how to downgrade to previous version ?

Thanks

**Comment by tiagoheineck** [comment]
> Is it using the WSL2 backend or the hyper-v one ?

i'm using hyper-v

**Comment by tiagoheineck** [comment]
> The error message says that you're out of disk space. Is that in fact the case? You could try `docker image prune` or `docker system prune` to clean up your disk space.

I ran the commands, cleaned everything, but it still doesn't work ...

the same happens on another machine

**Comment by mvegter** [comment]
Same here, had to uninstall 2.3.0.2 (45183) and install 2.2.0.5 (43884) to get everything up and running again.

**Comment by tiagoheineck** [comment]
> Hi,
> 
> I'have the same problem after update too (using Hyper-V).
> Not a real free disk space.
> 
> Someone can explain me how to downgrade to previous version ?
> 
> Thanks

Uninstall docker

Download previous version 2.2.0.5 here:
https://download.docker.com/win/stable/43884/Docker%20Desktop%20Installer.exe

After downgrade run ok!

**Comment by zikhan** [comment]
I'm having the same issue, however, I was able to get it working when I used the entire path for the host path.
```yaml
db:
    image: mariadb:10.3
    ports: 
    - "3306:3306"
    environment: 
        MYSQL_ROOT_PASSWORD: root
        MYSQL_DATABASE: root
        MYSQL_USER: root
        MYSQL_PASSWORD: root
    command: mysqld --innodb-flush-method=fsync
    volumes:
    - //c/some/dir/db:/var/lib/mysql 
```
There was a barely related mariadb docker image bug, but that's where I thought of the full host path: https://github.com/docker-library/mariadb/issues/138#issuecomment-378001714

Underneath it looks like innodb was trying to run posix_fallocate() but blows with that 38 error code.

**Comment by djs55** [comment]
While we're investigating this, you might be able to work around the problem and increase performance at the same time. Storing the database files on the host will always be slow, but if the main requirement is to guarantee persistence between runs then a `docker volume` will achieve that and be much faster (because the database files will be stored on the Linux filesystem rather on Windows where they have to be accessed remotely).

For example:
```
version: '3.3'
volumes:
  database:
services:
  db:
    image: mariadb:10.3.22
    ports: 
    - "3306:3306"
    environment: 
        MYSQL_ROOT_PASSWORD: covid
        MYSQL_DATABASE: plataformasaude
        MYSQL_USER: plataformasaude
        MYSQL_PASSWORD: covid     
    volumes:
    - database:/var/lib/mysql  
```

**Comment by Gael101** [comment]
I am having the same issue, therefore the issue only occurs on windows systems. On OSX systems everything runs smoothly.
I tried juggling with the entire set of options available with bind mounts, and nothing works on windows :/
Here are the logs related to the error.

`         | 2020-05-14 10:17:44+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 1:10.4.13+maria~bionic started.
db            | 2020-05-14 10:17:45+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
db            | 2020-05-14 10:17:45+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 1:10.4.13+maria~bionic started.
db            | 2020-05-14 10:17:45+00:00 [Note] [Entrypoint]: Initializing database files
db            | 2020-05-14 10:17:45 0 [ERROR] InnoDB: The Auto-extending innodb_system data file './ibdata1' is of a different size 0 pages than specified in the .cnf file: initial 768 pages, max 0 (relevant if non-zero) pages!
db            | 2020-05-14 10:17:45 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
db            | 2020-05-14 10:17:46 0 [ERROR] Plugin 'InnoDB' init function returned error.
db            | 2020-05-14 10:17:46 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
db            | 2020-05-14 10:17:46 0 [ERROR] Unknown/unsupported storage engine: InnoDB
db            | 2020-05-14 10:17:46 0 [ERROR] Aborting`

**Comment by mat007** [comment]
Thank you for your patience, we now have a fix for this issue in a private build for anyone who wants to try it: https://download-stage.docker.com/win/stable/45351/Docker%20Desktop%20Installer.exe

**Comment by zikhan** [comment]
just tried it. `fallocate -x -l 12M ibtmp1` works in the FUSE mount and Mariadb starts up!

**Comment by mat007** [comment]
The fix has now just been released in Docker Desktop Stable 2.3.0.3, I’ll go ahead and close this issue.
